.circleci/config.yml

```diff
@@ -15,152 +42,362 @@ references:
+              echo "Executing a PR based analysis"
+              mvn -s .circleci/.circleci.settings.xml sonar:sonar \
+                  -Dsonar.projectKey=<< pipeline.parameters.MODULE_ID >> \
+                  -Dsonar.pullrequest.branch=$CIRCLE_BRANCH \
+                  -Dsonar.pullrequest.key=${CIRCLE_PULL_REQUEST##*/} \
+                  -Dsonar.pullrequest.base=<< pipeline.parameters.PRIMARY_RELEASE_BRANCH >> \
+                  -Dsonar.pullrequest.github.repository=<< pipeline.parameters.GITHUB_SLUG >>
+            elif [[ "$CIRCLE_BRANCH" == << pipeline.parameters.PRIMARY_RELEASE_BRANCH >> ]]; then
+              echo "Executing an analysis on the main branch"
+              mvn -s .circleci/.circleci.settings.xml dependency-check:aggregate sonar:sonar \
+                  -Dsonar.projectKey=<< pipeline.parameters.MODULE_ID >> $DEPENDENCY_CHECK_SETTINGS
+            else
+              echo "Executing an analysis on branch: $CIRCLE_BRANCH"
+              mvn -s .circleci/.circleci.settings.xml dependency-check:aggregate sonar:sonar \
+                  -Dsonar.branch.name=$CIRCLE_BRANCH $DEPENDENCY_CHECK_SETTINGS
+            fi
       - save_cache:
           paths:
             - ~/.m2
-          key: v2-dependencies-{{ checksum "pom.xml" }}
+          key: sonar-dependencies-{{ checksum "pom.xml" }}
 
-# Pipeline parameter must be declared
-parameters:
-  run_pull_request:
-    type: boolean
-    default: false
-  target_branch:
-    type: string
-    default: ""
-  pull_request_number:
-    type: string
-    default: ""
-  github_repository:
-    type: string
-    default: ""
+  generate-changelog:
+    <<: *workdir
+    docker: # run the steps with Docker
+      - image: cimg/node:14.15
+    resource_class: xlarge
+    steps:
+      - checkout
+      - *attach-workspace
+      - add_ssh_keys:
+          fingerprints:
+            - "b9:ff:f3:58:b0:d7:e2:fa:9e:ca:e8:04:c9:d8:4f:f4"
+      - run:
+          name: Setting up github configuration
+          command: |
+            mkdir -p ~/.ssh
+            ssh-keyscan github.com >> ~/.ssh/known_hosts
+            git config user.email "jahia-ci@jahia.com"
+            git config user.name "Jahia CI"
+      - run:
+          name: Getting latest changes from the remote branch
+          command: |
+            git fetch --all
+            git pull origin $CIRCLE_BRANCH --quiet
+      - run:
+          name: Update Changelog
+          command: |
+            yarn
+            yarn add auto
+            GH_TOKEN=${GITHUB_API_TOKEN} yarn auto changelog
+            git push --set-upstream origin $CIRCLE_BRANCH
+      - *persist-workspace
+
+  release:
+    <<: *workdir
+    docker: # run the steps with Docker
+      - image: cimg/openjdk:8.0.275-node
+    resource_class: xlarge
+    steps:
+      - checkout
+      - add_ssh_keys:
+          fingerprints:
+            - "9b:1c:a7:f2:28:a3:fa:c7:c6:0b:a5:d3:e8:58:6c:15"
+      - run:
```

Comment:
you are missing this part
https://github.com/Jahia/jahia-authentication/commit/96a3d52792ba92dd0a6ec49a85bf2ac70b50f875#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R198-R202
in the release job